### PR TITLE
feat(RHTAPREL-172): rename pipelines

### PR DESCRIPTION
**Pull request by scoheb**
- rename of `release` -> `push-to-external-registry`
  - pipeline to push snapshots to quay ***with no*** pyxis interactions
- rename `push-to-external-registry` -> `rh-push-to-external-registry`
  - pipeline to push snapshots to quay ***with*** pyxis interactions
- add .gitignore

This supercedes #301

### CODE REVIEW

**Comment by scoheb** [comment]
Closing in favour of #302

**Comment by scoheb** [comment]
FYI: Being testing via https://github.com/redhat-appstudio/tenants-config/pull/123

**Review by mmalina — DISMISSED** [review]
Hmm, it would have been easier to review if those were two commits:
- push-to-external-registry -> rh-push-to-external-registry
- release -> push-to-external-registry

Because with one commit Git/Github is confused and thinks that push-to-external-registry was just modified and release was moved to rh-push-to-external-registry

But you would probably have to do it again. And then we have the mandatory squash here anyway, but it would probably make sense to disable it for this. With separate commit, the history would be better preserved.

File: .gitignore
Comment (johnbieren) [inline .gitignore:1]: I really don't think intelliJ stuff should be part of a commit about renaming pipelines

Comment (scoheb) [reply .gitignore:1]: fixed.

File: pipelines/push-to-external-registry/README.md [hunk 1]
```diff
@@ -13,123 +13,6 @@ Tekton pipeline to push images to an external registry.
 | enterpriseContractPolicy | JSON representation of the policy to be applied when validating the enterprise contract | No | - |
 | enterpriseContractPublicKey | Public key to use for validation by the enterprise contract | Yes | k8s://openshift-pipelines/public-key |
 | postCleanUp | Cleans up workspace after finishing executing the pipeline | Yes | true |
-| verify_ec_task_bundle | Switch back to using bundle resolvers for the verify-enterprise-contract task | No | - |
+| verify_ec_task_bundle | The location of the bundle containing the verify-enterprise-contract task | No | - |
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | Yes | main |
-
```
[the remark this answers is not in the document]
Comment (mmalina) [reply pipelines/push-to-external-registry/README.md:19]: Good point. I think the changelogs should be copied from the original task and then the new change would that this was renamed from `release` to `push-to-external-registry` and similarly with the other one.

**Comment by johnbieren** [comment]
Can we target the `development` branch for this instead of main?

**Comment by scoheb** [comment]
> Can we target the `development` branch for this instead of main?

done

File: pipelines/push-to-external-registry/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (scoheb) [reply pipelines/push-to-external-registry/README.md:19]: fixed

File: pipelines/rh-push-to-external-registry/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+| Name | Description | Optional | Default value |
+|------|-------------|----------|---------------|
+| release | The namespaced name (namespace/name) of the Release custom resource initiating this pipeline execution | No | - |
+| releaseplan | The namespaced name (namespace/name) of the releasePlan | No | - |
+| releaseplanadmission | The namespaced name (namespace/name) of the releasePlanAdmission | No | - |
+| snapshot | The namespaced name (namespace/name) of the snapshot | No | - |
+| enterpriseContractPolicy | JSON representation of the policy to be applied when validating the enterprise contract | No | - |
+| enterpriseContractPublicKey | Public key to use for validation by the enterprise contract | Yes | k8s://openshift-pipelines/public-key |
+| postCleanUp | Cleans up workspace after finishing executing the pipeline | Yes | true |
+| verify_ec_task_bundle | Switch back to using bundle resolvers for the verify-enterprise-contract task | No | - |
```
[the remark this answers is not in the document]
Comment (scoheb) [reply pipelines/rh-push-to-external-registry/README.md]: what is wrong? what should it be?

I see:

> Tekton pipeline to release Red Hat Snapshots to an external registry. This pipeline includes pyxis interactions.

**Review by mmalina — DISMISSED** [review]

File: pipelines/rh-push-to-external-registry/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnbieren) [reply pipelines/rh-push-to-external-registry/README.md]: `Switch back to using bundle resolvers for the verify-enterprise-contract task` should be `The location of the bundle containing the verify-enterprise-contract task`

[the remark this answers is not in the document]
Comment (scoheb) [reply pipelines/rh-push-to-external-registry/README.md]: fixed

**Review by johnbieren — APPROVED** [review]

**Pull request merged**